File: Skornel02.ETR.Common/Dtos/ClassRoomStatisticsDto.cs
namespace Skornel02.ETR.Common.Dtos;

public record ClassRoomStatisticsDto(
    int Capacity,
    List<ClassRoomStatisticsCourse> Courses,
    List<ClassRoomStatisticsExam> Exams
);

public record ClassRoomStatisticsCourse(
    string CourseCode,
    string CourseSemester
);

public record ClassRoomStatisticsExam(
    string CourseCode,
    string CourseSemester,
    DateTime ExamDate
);

File: Skornel02.ETR.Web/src/schemas/ClassRoomStatisticsDto.ts
import { z } from 'zod';

const ClassRoomStatisticsCourseSchema = z.object({
	courseCode: z.string(),
	courseSemester: z.string()
});

export type ClassRoomStatisticsCourse = z.infer<typeof ClassRoomStatisticsCourseSchema>;

const ClassRoomStatisticsExamSchema = z.object({
	courseCode: z.string(),
	courseSemester: z.string(),
	examDate: z.string()
});

export type ClassRoomStatisticsExam = z.infer<typeof ClassRoomStatisticsExamSchema>;

export const ClassRoomStatisticsDtoSchema = z.object({
	capacity: z.number(),
	courses: z.array(ClassRoomStatisticsCourseSchema),
	exams: z.array(ClassRoomStatisticsExamSchema)
});

export type ClassRoomStatisticsDto = z.infer<typeof ClassRoomStatisticsDtoSchema>;
